tests/test-titlebar.lua

```diff
@@ -0,0 +1,94 @@
+local runner = require("_runner")
+local titlebar = require("awful.titlebar")
+local rules = require("awful.rules")
+local spawn = require("awful.spawn")
+
+local tiny_client = {"lua", "-e", [[
+local Gtk, class = require('lgi').require('Gtk'), 'client'
+Gtk.init()
+window = Gtk.Window {default_width=100, default_height=100, title='title'}
+window:set_wmclass(class, class)
+local app = Gtk.Application {}
+function app:on_activate()
+    window.application = self
+    window:show_all()
+end
+app:run {''}
+]]}
+
+-- Use the test client props
+rules.rules = {}
+
+-- Too bad there's no way to disconnect the rc.lua request::titlebars function
```

Comment:
This could be either disconnected if it was set as a function (so its reference could be used), or with something that would allow to unset any signal handlers (for a given signal).
For the 2nd idea, it could provide a mechanism to restore them later, useful for tests and maybe even in general, if you want to execute something without any signals?!